### DISCUSSION TITLE
Suggestion: Persist language selection

### DISCUSSION THREAD

**Discussion by schalkneethling** [mdn/mdn-community#97] [MDN Plus]
Quoting @mt-webdev:

> Unfortunately, the language selection is not being persisted, which means on every visit I have to reselect my wanted language (which is English)
> ## What is the problem?
> The site is currently not able to persist in the user/session settings.
> ## What did you expect to happen?
> Once I revisit the site I would like to memorize my language selection ( & theme as well, but less of a bummer)

Also quoting @NHerw:

> Now that we also have MDN Plus in german speaking countries, can we save our preferred language in our mozilla account? -  https://github.com/mdn/yari/issues/5498#issuecomment-1114576876

The full conversation is here: https://github.com/mdn/yari/issues/5498 - Please be sure to make all future comments in this discussion thread. Thank you. - ping @mdn/core-dev

**Comment by bershanskiy** [comment]
I agree that the lack of language "memory" is inconvenient.

Advice to other users: you can install a browser extension that redirects/rewrites URLs to fix this. Just set up extension to replace the locale string in the URL with the desired one.

To Yari/MDN developers: I can prepare a PR with this feature if you can draw a low-fidelity sketch of UI or even just describe it in plain words.

**Reply by schalkneethling** [reply]
Thank you so much for your offer. I will highlight this to engineering and get back to you.

**Comment by caugner** [comment]
We're resolving this problem with the ["Remember language" experiment](https://github.com/orgs/mdn/discussions/739). 🎉
